File: 🟢 Easy/1957 - Delete Characters to Make Fancy String/_1957DeleteCharacterstoMakeFancyString.java
class Solution {
    public String makeFancyString(String s) {
        int count = 1;
        char prev = s.charAt(0);
        StringBuilder res = new StringBuilder();
        res.append(s.charAt(0));
        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i) == prev) {
                count++;
            } else {
                prev = s.charAt(i);
                count = 1;
            }
            if (count < 3) {
                res.append(s.charAt(i));
            }
        }
        return res.toString();
    }
}

File: 🟢 Easy/1957 - Delete Characters to Make Fancy String/_1957DeleteCharacterstoMakeFancyString.js
/**
 * @param {string} s
 * @return {string}
 */
var makeFancyString = function (s) {
    let prev = s[0];
    let count = 1;
    let res = s[0];

    for (let i = 1; i < s.length; i++) {
        if (s.charAt(i) === prev) {
            count++;
        } else {
            count = 1;
            prev = s.charAt(i);
        }

        if (count < 3) {
            res += s.charAt(i);
        }
    }

    return res;
};